input/blog/2025-07-17-dotnet-cake-cs.md

```diff
@@ -0,0 +1,296 @@
+
+Cake.Sdk is built on top of **Cake.Generator**, which provides automatic source generation capabilities. This means you get all the benefits of Cake's powerful aliases and functionality without any additional configuration.
+
+### What's Included
+
+The Cake.Sdk automatically configures the following properties:
+
+- `OutputType`: Exe
+- `Nullable`: enable
+- `ImplicitUsings`: enable
+- `Optimize`: true
+- `DebugType`: portable
+- `DebugSymbols`: true
+- `LangVersion`: latest
+- `PublishAot`: false
+- `JsonSerializerIsReflectionEnabledByDefault`: true
+
+### Default Package References
+
+The following packages are automatically included:
+
+- [Cake.Generator](https://www.nuget.org/packages/Cake.Generator) - Source generator for Cake aliases
+- [Cake.Core](https://www.nuget.org/packages/Cake.Core) - Core Cake functionality
+- [Cake.Common](https://www.nuget.org/packages/Cake.Common) - Core Common functionality
+- [Cake.Cli](https://www.nuget.org/packages/Cake.Cli) - Command-line interface for Cake
+- [Cake.DotNetTool.Module](https://www.nuget.org/packages/Cake.DotNetTool.Module) - .NET Tool support for Cake
+- [Cake.NuGet](https://www.nuget.org/packages/Cake.NuGet) - NuGet support for Cake
+- [Microsoft.Extensions.DependencyInjection](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection) - IoC container support
+
+### Cake Addin & Module Support
+
+Just like with Cake tool scripting, code is generated for having methods, properties, and models conveniently available automatically by just adding NuGet package references. The same applies to modules, enabling you to replace Cake core functionality just by adding a reference to a Cake module.
+
+#### Example Cake Addin in File-based Project
+
+```csharp
+#:sdk Cake.Sdk
+#:package Cake.Twitter@5.0.0
+
+var oAuthConsumerKey        = EnvironmentVariable("TWITTER_CONSUMER_KEY");
+var oAuthConsumerSecret     = EnvironmentVariable("TWITTER_CONSUMER_SECRET");
+var accessToken             = EnvironmentVariable("TWITTER_ACCESS_TOKEN");
+var accessTokenSecret       = EnvironmentVariable("TWITTER_ACCESS_TOKEN_SECRET");
+
+TwitterSendTweet(oAuthConsumerKey, oAuthConsumerSecret, accessToken, accessTokenSecret, "Testing, 1, 2, 3");
+```
+
+Any Cake addin can be added as a `PackageReference` and its alias proxies will be generated automatically. The generator will scan the addin assembly and create proxy methods for all discovered Cake method and property aliases, making them available as static methods without requiring explicit `ICakeContext` parameters.
+
+#### Example Module
+
+```csharp
+#:sdk Cake.Sdk
+#:package Cake.BuildSystems.Module@8.0.0
+```
+
+The generator now supports Cake modules with automatic registration. Modules referenced in your project will have their method and property aliases automatically discovered and proxy methods generated, just like regular addins. This includes both NuGet package modules and local module assemblies.
```

Comment:
Since this is the announcement, it does not make sense to add `now`, since there was no before for the reader :-)

```suggestion
The generator supports Cake modules with automatic registration. Modules referenced in your project will have their method and property aliases automatically discovered and proxy methods generated, just like regular addins. This includes both NuGet package modules and local module assemblies.
```